EIPS/eip-5007.md

```diff
@@ -0,0 +1,313 @@
+        require(_isApprovedOrOwner(_msgSender(), firstTokenId) &&  _isApprovedOrOwner(_msgSender(), secondTokenId),
+          "error: caller is not owner nor approved");
+
+        TimeNftInfo memory firstToken = _timeNftMapping[firstTokenId];
+        TimeNftInfo memory secondToken = _timeNftMapping[secondTokenId];
+
+        require(firstToken.originalTokenId == secondToken.originalTokenId 
+                && firstToken.startTime <= firstToken.endTime 
+                && (firstToken.endTime + 1) == secondToken.startTime
+                && secondToken.startTime <= secondToken.endTime, "invalid tokenId");
+
+        _burn(firstTokenId);
+        _burn(secondTokenId);
+
+        newTokenId = _mintTimeNft(newTokenOwner, firstToken.originalTokenId, firstToken.startTime, secondToken.endTime);
+    }
+
+    /// @notice mint a new time NFT
+    /// @param to_  The owner of the new token
+    /// @param originalTokenId_  The token id which the new token mint from
+    /// @param startTime_  The start time of the new token
+    /// @param endTime_  The end time of the new token
+    /// @return newTokenId  The id of the new token
+    function _mintTimeNft(address to_, uint256 originalTokenId_, uint64 startTime_, uint64 endTime_) internal virtual returns(uint256 newTokenId)  {
+        newTokenId = _nextTokenId;
+        _nextTokenId++;
+
+        TimeNftInfo storage info = _timeNftMapping[newTokenId];
+        info.originalTokenId = originalTokenId_;
+        info.startTime = startTime_;
+        info.endTime = endTime_;
+
+        _mint(to_, newTokenId);
+    }
+
+    /// @notice mint a new original time NFT
+    /// @param to_  The owner of the new token
+    /// @param startTime_  The start time of the new token
+    /// @param endTime_  The end time of the new token
+    /// @return newTokenId The id of the new token
+    function _mintOriginalToken(address to_, uint64 startTime_, uint64 endTime_) internal virtual returns(uint256 newTokenId) {
+        newTokenId = _nextTokenId;
+        _nextTokenId++;
+    
+        TimeNftInfo storage info = _timeNftMapping[newTokenId];
+        info.originalTokenId = newTokenId;
+        info.startTime = startTime_;
+        info.endTime = endTime_;
+
+        _mint(to_, newTokenId );
+    }
+
+    /// @notice burn a time NFT
+    /// @param tokenId  The id of the token
+    function _burn(uint256 tokenId) internal  virtual override{
+        super._burn(tokenId);
+        delete _timeNftMapping[tokenId];
+    }
+
+    /// @dev See {IERC165-supportsInterface}.
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+        return interfaceId == type(IERC5007).interfaceId || super.supportsInterface(interfaceId);
+    }
+}
+
+```
+
+## Security Considerations
+
+Implementors of the `ERC5007` standard must consider the condition of `merge` function.  The `originalTokenId` of two NFTs should be same and the time of two NFTs should be adjacent.
```

Comment:
I don't understand what this means.

Reply:
> 

I will add more test code later.

Reply:
I meant more the English description of the security consideration. I don't understand what you're trying to convey. I think it just needs to be reworded a bit to be more understandable.